smarts/core/smarts.py

```diff
@@ -900,8 +900,13 @@ def _pybullet_provider_sync(self, provider_state: ProviderState):
                     )
                 # Update the social vehicle avatar to match the vehicle state
                 if not vehicle.updated:
-                    # Note:  update_state() happens *after* pybullet has been stepped.
-                    social_vehicle.update_state(vehicle, dt=dt)
+                    if self._resetting:
```

Comment:
Actually, I now realise that there is still a potential problem here because the lack of physics update is caused by having no active agents. Just fixing this by updating during reset leaves open the possibility of it happening in the multi-agent case where agent become active in a way that leaves a gap in the scenario where there are no agent.

We might want this case to be based on there not being an active agent.

Reply:
Changed `if self._resetting:` to `if not self._vehicle_index.agent_vehicle_ids():` .